src/lib/containers/drug_upload_tool/UserFileGrid.tsx

```diff
@@ -159,6 +150,47 @@ export default class UserFileGrid extends React.Component<
       })
     }
   }
+
+  getMore = async (fileListType: FileListType, nextPageToken: string) => {
+    const statusList =
+      fileListType === 'SUBMITTED'
+        ? UserFileGrid.requestFilter.SUBMITTED
+        : UserFileGrid.requestFilter.IN_PROGRESS
+    const result = await this.getTypeFileListing(statusList, nextPageToken)
+    if (fileListType === 'SUBMITTED') {
+      this.setState(prevState => ({
+        submitted: {
+          fileList: [...prevState.submitted.fileList, ...result.fileList],
+          nextPageToken: result.nextPageToken,
+        },
+      }))
+    } else {
+      this.setState(prevState => ({
+        inProgress: {
+          fileList: [...prevState.inProgress.fileList, ...result.fileList],
+          nextPageToken: result.nextPageToken,
+        },
+      }))
+    }
+  }
+
+  getUserFileListing = async (): Promise<void> => {
+    const inProgress = await this.getTypeFileListing(
```

Comment:
These two calls could be changed so that they are run in parallel (no dependency on one another) to reduce the time it takes to see submitted forms.